app/javascript/src/NewService/index.jsx

```diff
@@ -0,0 +1,127 @@
+// @flow
+
+import React, {useState, useEffect} from 'react'
+import {fetch as fetchPolyfill} from 'whatwg-fetch'
+import 'core-js/es6/promise'
+
+import ServiceSourceForm from 'NewService/components/ServiceSourceForm'
+import ServiceDiscoveryForm from 'NewService/components/ServiceDiscoveryForm'
+import ServiceNewForm from 'NewService/components/ServiceNewForm'
+
+const BASE_URL = '/p/admin/service_discovery/'
+const NAMESPACES_URL = `${BASE_URL}projects.json`
+
+import type {SourceFormType, ServiceDiscoveryFormType, ServiceNewFormType} from 'NewService/types'
+
+type Props = {
+  isServiceDiscoveryAccesible: boolean,
+  isServiceDiscoveryUsable: boolean,
+  serviceDiscoveryAuthenticateUrl: string,
+  providerAdminServiceDiscoveryServicesPath: string,
+  adminServicesPath: string
+}
+
+const fetchData = async (url: string) => {
+  const dataItem = url === NAMESPACES_URL
+    ? 'projects'
+    : 'services'
+
+  return fetchPolyfill(url).then(response => {
+    if (!response.ok) {
+      throw Error(response.statusText)
+    }
+    return response.json()
+  })
+    .then(data => data[dataItem].reduce((acc, current) => acc.concat(current.metadata.name), []))
+    .catch(error => console.error(error))
+}
+
+// eslint-disable max-lines-per-function
+const NewServiceWrapper = (props: Props) => {
+  const [isServDescVisible, setServDescVisible] = useState(false)
+  const [isServNewVisible, setServNewVisible] = useState(true)
+  const [namespaces, setNamespaces] = useState([])
+  const [services, setServices] = useState([])
+  const [selectedNamespace, setSelectedNamespace] = useState(undefined)
+  const [servicesRequested, setServicesRequested] = useState(false)
+
+  const settingSelectedNamespace = (namespace: string) => {
+    setSelectedNamespace(namespace)
+  }
+
+  const onChangeNamespaces = (event: SyntheticEvent<HTMLSelectElement>) => {
+    setSelectedNamespace(event.currentTarget.value)
+    setServicesRequested(true)
+  }
+  // eslint-disable-next-line flowtype/no-weak-types
+  const startNamespacesFetch = async (): Promise<any> => {
+    const namespacesData = await fetchData(NAMESPACES_URL)
+    if (namespacesData) {
+      setNamespaces(namespacesData)
+      setServicesRequested(true)
+      settingSelectedNamespace(namespacesData[0])
+    }
+  }
+
+  const startServicesFetch = async (servicesUrl: string) => {
```

Comment:
`startServiceFetch` implies it's a sync method that it will eventually end. It doesn't return anything either so I wouldn't call it fetch or call. `refresh` or `update` could be a better name.